File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/Fascicles/FascicleModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using VecompSoftware.Commons.Interfaces.CQRS.Commands;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;
using VecompSoftware.DocSuiteWeb.Model.Entities.DocumentUnits;
using VecompSoftware.DocSuiteWeb.Model.Entities.Dossiers;
using VecompSoftware.DocSuiteWeb.Model.Entities.Tenants;
using VecompSoftware.DocSuiteWeb.Model.Processes;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.Fascicles
{
    public class FascicleModel : IContentBase
    {
        #region [ Constructors ]

        public FascicleModel()
        {
            Contacts = new List<ContactModel>();
            FascicleDocuments = new List<FascicleDocumentModel>();
            FascicleDocumentUnits = new List<FascicleDocumentUnitModel>();
            FascicleRoles = new List<FascicleRoleModel>();
            DocumentUnits = new List<DocumentUnitModel>();
            FascicleFolders = new List<FascicleFolderModel>();
            DossierFolders = new List<DossierFolderModel>();
        }

        #endregion

        #region [ Properties ]
        public short? Conservation { get; set; }
        public DateTimeOffset? EndDate { get; set; }
        public string FascicleObject { get; set; }
        public FascicleType? FascicleType { get; set; }
        public VisibilityType? VisibilityType { get; set; }
        public string Manager { get; set; }
        public string Name { get; set; }
        public string Note { get; set; }
        public int? Number { get; set; }
        public string Rack { get; set; }
        public DateTimeOffset? StartDate { get; set; }
        public string Title { get; set; }
        public short? Year { get; set; }
        public Guid UniqueId { get; set; }
        public string RegistrationUser { get; set; }
        public DateTimeOffset? RegistrationDate { get; set; }
        public DateTimeOffset? LastChangedDate { get; set; }
        public CategoryModel Category { get; set; }
        public string MetadataDesigner { get; set; }
        public string MetadataValues { get; set; }
        public int? DSWEnvironment { get; set; }
        public string CustomActions { get; set; }
        public string ProcessLabel { get; set; }
        public string DossierFolderLabel { get; set; }
        public MetadataRepositoryModel MetadataRepository { get; set; }
        public ContactModel ManagerC { get; set; }
        public ICollection<ContactModel> Contacts { get; set; }
        public ICollection<FascicleDocumentUnitModel> FascicleDocumentUnits { get; set; }
        public ICollection<FascicleDocumentModel> FascicleDocuments { get; set; }
        public ICollection<FascicleRoleModel> FascicleRoles { get; set; } 
        public ICollection<DocumentUnitModel> DocumentUnits { get; set; }
        public ContainerModel Container { get; set; }
        public ICollection<FascicleFolderModel> FascicleFolders { get; set; }
        public TenantAOOModel TenantAOO { get; set; }
        public ICollection<DossierFolderModel> DossierFolders { get; set; }
        public ProcessFascicleTemplateModel FascicleTemplate { get; set; }

        #endregion

        #region [ Methods ]

        public bool HasContacts()
        {
            return Contacts.Any();
        }

        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Fascicles/FascicleModel.ts

import FascicleType = require('App/Models/Fascicles/FascicleType');
import VisibilityType = require('App/Models/Fascicles/VisibilityType');
import FascicleDocumentUnitModel = require('App/Models/Fascicles/FascicleDocumentUnitModel');
import FascicleLinkModel = require('App/Models/Fascicles/FascicleLinkModel');
import FascicleRoleModel = require('App/Models/Fascicles/FascicleRoleModel');
import CategoryModel = require('App/Models/Commons/CategoryModel');
import ContactModel = require('App/Models/Commons/ContactModel');
import FascicleDocumentModel = require('App/Models/Fascicles/FascicleDocumentModel');
import MetadataRepositoryModel = require('App/Models/Commons/MetadataRepositoryModel');
import FascicleFolderModel = require('App/Models/Fascicles/FascicleFolderModel');
import ContainerModel = require('App/Models/Commons/ContainerModel');
import DossierFolderModel = require('App/Models/Dossiers/DossierFolderModel');
import ProcessFascicleTemplateModel = require('App/Models/Processes/ProcessFascicleTemplateModel');
import WorkflowActionModel = require('App/Models/Workflows/WorkflowActionModel');
import TenantAOOModel = require('App/Models/Tenants/TenantAOOModel');

class FascicleModel {
    UniqueId: string;
    Year: number;
    Number: number;
    Conservation: number;
    Manager: string;
    StartDate: Date;
    EndDate: Date;
    Rack: string;
    Name: string;
    FascicleObject: string;
    Note: string;
    Title: string;
    RegistrationUser: string;
    RegistrationDate: Date;
    LastChangedUser: string;
    LastChangedDate: Date;
    FascicleType: FascicleType;
    VisibilityType: VisibilityType;
    DSWEnvironment: number;
    MetadataDesigner: string;
    MetadataValues: string;
    Category: CategoryModel;
    Container: ContainerModel;
    MetadataRepository: MetadataRepositoryModel;
    FascicleTemplate: ProcessFascicleTemplateModel;
    Contacts: ContactModel[];
    FascicleDocumentUnits: FascicleDocumentUnitModel[];
    FascicleLinks: FascicleLinkModel[];
    LinkedFascicles: FascicleLinkModel[];
    FascicleDocuments: FascicleDocumentModel[];
    FascicleRoles: FascicleRoleModel[];
    FascicleFolders: FascicleFolderModel[];
    DossierFolders: DossierFolderModel[];
    WorkflowActions: WorkflowActionModel[];
    CustomActions: string;
    TenantAOO: TenantAOOModel

    /**
     * Costruttore
     */
    constructor() {
        this.Contacts = new Array<ContactModel>();
        this.FascicleDocumentUnits = new Array<FascicleDocumentUnitModel>();
        this.FascicleLinks = new Array<FascicleLinkModel>();
        this.LinkedFascicles = new Array<FascicleLinkModel>();
        this.FascicleDocuments = new Array<FascicleDocumentModel>();
        this.FascicleRoles = new Array<FascicleRoleModel>();
        this.FascicleFolders = new Array<FascicleFolderModel>();
        this.WorkflowActions = new Array<WorkflowActionModel>();
    }
}

export = FascicleModel;
